### ISSUE TITLE
Incorrect guess from data

### ISSUE THREAD

**Issue by nebulousGirl** [Gregwar/Image#99]
You could use getimagesizefromstring to guess type from Data.

**Comment by Gregwar** [comment]
Hello,
Do you have any example of use case?
Would you like to make a PR?

**Comment by nebulousGirl** [comment]
Yes, I will make a PR when I have the time and sorry for the lack of details. 

I created the issue mainly as reminder to make a PR.

Mainly, the problem is that when Source/Data is used it always return jpeg when guessing the type.

**Comment by nebulousGirl** [comment]
Any chance to see this merged?
